app/views/admin/admin_users/_user_form.html.erb

```diff
@@ -4,11 +4,15 @@
   <%= hidden_field_tag :query, params[:query] %>
   <%= hidden_field_tag :role, params[:role] %>
   <%= hidden_field_tag "id", user.login %>
+  <% # HACK: We need the user param to be present so we can remove all roles. 
+     # However, if we simply submit the form with all roles unchecked, the user
+     # param is stripped. Therefore, we need a placeholder. %>
+  <%= hidden_field_tag "user[roles][]", [""], disabled: !admin_can_update_user_roles? %>
```

Comment:
`user[roles][]` is the part specifying the array, you can use `""` instead of `[""]` for the value.

Reply:
Ah, okay! I thought that wouldn't work because it left the value attribute totally blank in the HTML (like not even `value=""` -- just `value`), but it does indeed seem to work.